qt/applications/workbench/workbench/plotting/globalfiguremanager.py

```diff
@@ -97,19 +117,13 @@ def destroy(cls, num):
         """
         if not cls.has_fignum(num):
             return
-        manager = cls.figs[num]
-        manager.canvas.mpl_disconnect(manager._cidgcf)
+        current_fig_manager = cls.figs[num]
+        current_fig_manager.canvas.mpl_disconnect(current_fig_manager._cidgcf)
 
-        # There must be a good reason for the following careful
-        # rebuilding of the activeQue; what is it?
-        oldQue = cls._activeQue[:]
-        cls._activeQue = []
-        for f in oldQue:
-            if f != manager:
-                cls._activeQue.append(f)
+        cls.remove_manager_if_present(current_fig_manager)
```

Comment:
I'd like some feedback on this - do you think this is a good change? The comment leads to believe that something more is being done - but it seems the list is simply being deep copied without the current active manager.